test/e2e/tests/commons/sinspqa/sinsp.py

```diff
@@ -132,22 +132,17 @@ def validate_event(expected_fields: dict, event: dict) -> bool:
     Returns:
         True if all `expected_fields` are in the event and have matching values, False otherwise.
     """
-    if event is None:
-        return False
+    assert event is not None, f"Did not get expected event {expected_fields}"
 
     for k in expected_fields:
-        if k not in event:
-            return False
+        assert k in event, f"Event {event} does not have field {k}: expected {expected_fields}, got {event}"
 
         expected = expected_fields[k]
 
         if isinstance(expected, SinspField):
-            if expected.compare(str(event[k])):
-                continue
-            return False
-
-        if expected != event[k]:
-            return False
+            assert expected.compare(str(event[k])), f"Event {event} has unexpected value for {k}: expected {expected_fields}, got {event}"
+        else:
+            assert expected == event[k], f"Event {event} has unexpected value for {k}: expected {expected_fields}, got {event}"
```

Comment:
I'm pretty sure these changes will completely break the e2e tests, for instance a couple of non JSON lines are printed on startup mentioning the type of engine being used, those fail to be parsed and I'm pretty sure `None` is passed in here when that happens, which means the first assert will stop tests immediately.

Reply:
If you want to better inspect results, an HTML report is uploaded as a zip to the [summary of the run](https://github.com/falcosecurity/libs/actions/runs/3989054943), it lists the failing tests and provides the logs for `sinsp-example` as well as any other containers that run as part of tests.

Reply:
It should be obvious I have no idea what I'm doing here :D Is there a doc that describes how to run the e2e tests locally?

Reply:
There's a README on it 🙂: https://github.com/falcosecurity/libs/blob/master/test/e2e/README.md

The only caveat is that, because of the way the tests are built, `sinsp-example` needs to be built inside a `debian:buster` container and because it also builds the drivers, the kernel on the system should be buildable by that same image, so a debian buster or Ubuntu 20.04 VM is needed.

I know, that is really annoying and it should be possible to just run these on any type of system the libs are buildable from, I'll try to improve this if I get the time.

I also think @deepskyblue86 had some success running these by enabling them on his fork of the libs and creating PRs, so you don't need to be extra careful on commits and you can also disable things you don't need.

Reply:
Suggestions:
- @gnosek Could we revert those sinsp.py changes and only make the adjustments needed for the tests to pass, (read more below)?
- @Molter73 yes I also can't run the e2e tests on any of my machines, it's not just about ubuntu, but correct GLIBC version aka correct ubuntu :/ therefore a big +1 to refactor and use a centos7 container for the userspace builds ... a ubuntu container for the driver builds is fine.


Meanwhile, could we try to merge this PR? From my perspective the only changes needed to e2e would be explicitly adding the `-j` flag, e.g. here https://github.com/falcosecurity/libs/blob/97378b317e0ef8a811bb3a18bad9f6338d69cde0/test/e2e/tests/test_process/test_container.py#L6, because the old behavior was to default to JSON when using `-o` which is not the case anymore. @Molter73 wondering if you could help pointing us to all necessary changes in order to speed up the review process 🙏 thanks a bunch!